3.evaluate_model/scripts/nbconverted/class_PR_curves.py

```diff
@@ -88,27 +83,135 @@
         compiled_class_PR_curves.append(PR_data)
 
 
-# ### Save PR curves from each evaluation
+# ### Save PR curves from each evaluation (multiclass)
 # 
 
 # In[4]:
 
 
 # compile list of tidy data into one dataframe
-compiled_class_PR_curves = pd.concat(compiled_class_PR_curves).reset_index(drop=True)
+# some thresholds are None because last PR value doesnt correspond to cell dataset (these values are always P=1, R=0), remove these rows from PR data
+compiled_class_PR_curves = (
+    pd.concat(compiled_class_PR_curves).dropna().reset_index(drop=True)
+)
 
 # specify results directory
-class_PR_curves_dir = pathlib.Path("evaluations/class_precision_recall_curves/")
-class_PR_curves_dir.mkdir(parents=True, exist_ok=True)
+PR_curves_dir = pathlib.Path("evaluations/precision_recall_curves/")
+PR_curves_dir.mkdir(parents=True, exist_ok=True)
 
 # define save path
-compiled_scores_save_path = pathlib.Path(
-    f"{class_PR_curves_dir}/compiled_class_PR_curves.tsv"
+compiled_PR_data_save_path = pathlib.Path(
+    f"{PR_curves_dir}/compiled_class_PR_curves.tsv"
 )
 
 # save data as tsv
-compiled_class_PR_curves.to_csv(compiled_scores_save_path, sep="\t")
+compiled_class_PR_curves.to_csv(compiled_PR_data_save_path, sep="\t")
 
 # preview tidy data
 compiled_class_PR_curves
 
+
+# ### Evaluate each model on each dataset (multiclass)
+# 
+
+# In[5]:
+
+
+# directory to load the models from
+single_class_models_dir = pathlib.Path("../2.train_model/models/single_class_models")
+
+# use a list to keep track of scores in tidy long format for each model and dataset combination
+compiled_class_PR_curves = []
+
+# define combinations to test over
+model_types = ["final", "shuffled_baseline"]
+feature_types = ["CP", "DP", "CP_and_DP"]
+evaluation_types = ["train", "test"]
+phenotypic_classes = features_dataframe["Mitocheck_Phenotypic_Class"].unique()
+
+# colors for plotted lines
+data_split_colors = {
+    "CP_and_DP (test)": "#C214CB",
+    "CP_and_DP (train)": "#E88EED",
+    "CP (test)": "#EB4B4B",
+    "CP (train)": "#F8B5B5",
+    "DP (test)": "#5158bb",
+    "DP (train)": "#B5B9EA",
```

Comment:
Would it provide any benefit to consider the use of [colormaps](https://matplotlib.org/stable/tutorials/colors/colormaps.html) somehow for generating these colors?

Reply:
I think in the future this would be worth implementing. For now I think it's best to keep these colors the same as those established by gwaybio in #24